app/scripts/controllers/metametrics.js

```diff
@@ -0,0 +1,415 @@
+ * a user by being able to trace their activity on etherscan/block exploring
+ */
+const trackableSendCounts = {
+  1: true,
+  10: true,
+  30: true,
+  50: true,
+  100: true,
+  250: true,
+  500: true,
+  1000: true,
+  2500: true,
+  5000: true,
+  10000: true,
+  25000: true,
+}
+
+export function sendCountIsTrackable(sendCount) {
+  return Boolean(trackableSendCounts[sendCount])
+}
+
+/**
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsContext} MetaMetricsContext
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventPayload} MetaMetricsEventPayload
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventOptions} MetaMetricsEventOptions
+ * @typedef {import('../../../shared/constants/metametrics').SegmentEventPayload} SegmentEventPayload
+ * @typedef {import('../../../shared/constants/metametrics').SegmentInterface} SegmentInterface
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsPagePayload} MetaMetricsPagePayload
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsPageOptions} MetaMetricsPageOptions
+ */
+
+/**
+ * @typedef {Object} MetaMetricsControllerState
+ * @property {?string} metaMetricsId - The user's metaMetricsId that will be
+ *  attached to all non-anonymized event payloads
+ * @property {?boolean} participateInMetaMetrics - The user's preference for
+ *  participating in the MetaMetrics analytics program. This setting controls
+ *  whether or not events are tracked
+ * @property {number} metaMetricsSendCount - How many send transactions have
+ *  been tracked through this controller. Used to prevent attaching sensitive
+ *  data that can be traced through on chain data.
+ */
+
+export default class MetaMetricsController {
+  /**
+   * @param {boolean} isDevOrTestEnvironment - is the extension running in test
+   *  of dev mode? This will impact how segment is initialized
+   * @param {SegmentInterface} segmentMock - when in test or dev mode use this
+   *  mocked Segment module instead of an instance of Analytics
+   * @param {SegmentInterface} segmentLegacyMock - when in test or dev mode use
+   *  this mocked Segment module instead of an instance of Analytics for legacy
+   *  events
+   * @param {string} segmentWriteKey - the write key for the Segment source to
+   *  track new schema events to
+   * @param {string} segmentLegacyWriteKey - the write key for the Segment
+   *  source to tack legacy events to
+   * @param {string} segmentHost - the host to send events to. The default
+   *  host for Analytics constructor is 'https://api.segment.io'
+   * @param {number} flushAt - the maximum number of events to queue before
+   *  sending to Segment
+   * @param {number} flushInterval - the maximum number of milliseconds to wait
+   *  for before sending queued events
+   * @param {Object} preferencesController - The preferences controller, used
```

Comment:
One of the suggestions in the background controller refactor proposal was to limit what we pass into controllers to just what's required. So passing in just the store, or just the methods required, rather than the entire controller. The thought was that this would make the relationships between controllers more obvious, and would make unit tests easier to write because the mocks would be simpler.

Reply:
Agree.

Reply:
Q: Changing this to just pass in preferences store is easy, but doing the same for the network controller would I need to pass in the 'on' method as well as the various getters so that I can subscribe to the network did change event? @Gudahtt

Reply:
We dealt with a similar case here by early-binding `on` to the event we wanted to subscribe to: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L149

To avoid passing in the getters separately, we could setup the subscription entirely in `metamask-controller`, and pass in a handler that gets the `chainId` and network identifier directly :thinking: 

Not sure what I think about that. I wish we could use the store for all of our getters (via selector-like functions), but that might be awkward as things stand currently. Passing in the getters and subscribe function separately seems OK too - whatever works.

Reply:
I went with the easier approach of passing in the methods and subscribe method for now.